### ISSUE TITLE
Replaced ResNet50 to VGG16 as backbone

### ISSUE THREAD

**Issue by tusharsingh62** [Wanggcong/Spatial-Temporal-Re-identification#18]
I am working on this spatial-temporal person Reid. I have a constraint to use VGG16 as a backbone architecture and added/removed some customized layers at the back of VGG16 architecture. I wanted to improve my Rank@1 and mAP values on the Market-1501 dataset. I tried with my own visual stream features (VGG16) with spatial-temporal constraints. I got some results and would like to discuss it with you. Please see the screenshot that I have shared with you. I evaluated the VGG16 model with spatial-temporal constraint and without temporal-constraint. The interesting part is when I used spatial-temporal constraint Rank@1 metric is decreased by around 3%( Please see the values on screenshot). I'd like to discuss with you that why values gets decreased when applied spatial-temporal constraint and How can I improve it with using VGG16 architecture? 
![Screen Shot 2019-10-08 at 12 45 12](https://user-images.githubusercontent.com/28837107/66366299-e66bc980-e9ca-11e9-8e33-fbcb0313ea96.png)

**Comment by Wanggcong** [comment]
How do you compute the distance/similarity of VGG based features? Do you normalize the features? In our paper, I use cosine similarity.  I achieve this by normalizing the extracted features (in test_st_duke.py and test_st_market.py) like this

for i in range(2):
            if(i==1):
                img = fliplr(img)
            input_img = Variable(img.cuda())
            outputs = model(input_img) 
            f = outputs.data.cpu()
            ff = ff+f
        # norm feature
        if opt.PCB:
            # feature size (n,2048,4)
            fnorm = torch.norm(ff, p=2, dim=1, keepdim=True)
            ff = ff.div(fnorm.expand_as(ff))
            ff = ff.view(ff.size(0), -1)
        else:
            fnorm = torch.norm(ff, p=2, dim=1, keepdim=True)
            ff = ff.div(fnorm.expand_as(ff))

**Comment by tusharsingh62** [comment]
In the VGG16 case, the dimension of the features was 512. I use this cosine similarity and normalized features ( as mentioned in test_st_market.py) and I didn't use PCB.

**Comment by tusharsingh62** [comment]
I just want to ask you, If I changed the alpha and gamma values for training, would that be effective for VGG16 architecture? 
Alpha and smoothing factor is 5 and 50 respectively. These values are experimentally calculated if you consider ResNet50 architecture but for VGG16 it may be different. Is my approach is correct?

**Comment by Wanggcong** [comment]
Yes, alpha and smoothing factor could affect the performance. You can try different values for evaluation.

**Comment by Wanggcong** [comment]
As a simple debugging method, you can try this:
1) remove gaussian smoothing and only use histogram statistics;
2) replace this line (evaluation_st.py)
1/(1+np.exp(-alpha*score))*1/(1+2*np.exp(-alpha*score_st))
by
score  = score*(score_st>thelta), where thelta is an empirical threshold value. You can set a low value just for debugging.  

That means one directly removes some impossible images.

This simple method could improve the rank-1 accuracy but could harm the mAP.  I think this could be a simple debugging method for your case. But it is not a good method to improve both rank-1 accuracy and mAP.

**Comment by tusharsingh62** [comment]
Thanks a lot for your comments. I will try!
